extensions/wikia/AdEngine/js/utils/btfBlocker.js

```diff
@@ -0,0 +1,116 @@
+/*global define*/
+define('ext.wikia.adEngine.utils.btfBlocker', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.slotTweaker',
+	'wikia.lazyqueue',
+	'wikia.log',
+	'wikia.window'
+], function (adContext, slotTweaker, lazyQueue, log, win) {
+	'use strict';
+
+	var atfSlots = {
+			oasis: [
+				'CORP_TOP_LEADERBOARD',
+				'CORP_TOP_RIGHT_BOXAD',
+				'HOME_TOP_LEADERBOARD',
+				'HOME_TOP_RIGHT_BOXAD',
+				'HUB_TOP_LEADERBOARD',
+				'INVISIBLE_SKIN',
+				'TOP_LEADERBOARD',
+				'TOP_RIGHT_BOXAD',
+				'GPT_FLUSH'
+			],
+			mercury: [
+				'MOBILE_TOP_LEADERBOARD',
+				'INVISIBLE_HIGH_IMPACT'
+			]
+		},
+		btfQueue = [],
+		btfQueueStarted = false,
+		fillInSlot,
+		logGroup = 'ext.wikia.adEngine.utils.btfBlocker',
+		pendingAtfSlots = []; // ATF slots pending for response
+
+	function init(skin, fillInSlotCallback) {
+		log(['init', skin], 'debug', logGroup);
+
+		fillInSlot = fillInSlotCallback;
+		atfSlots = atfSlots[skin];
+
+		// There is no win.ads in Mercury and we already use win.ads.runtime.disableBtf variable on Oasis.
+		// Let's make it easy for AdOps - let's make the one way to disabling BTF on both skins.
+		// The following code sets the variable to false by default.
+		if (skin === 'mercury') {
+			win.ads = {
+				runtime: {
+					disableBtf: false
+				}
+			};
```

Comment:
It will override whole object if it will be needed in future. What about:
```
win.ads = win.ads || {};
win.ads.runtime = win.ads.runtime || {};
win.ads.runtime.disableBtf = false;
```
No matter what skin is used.